server/Models/SurrenderApplication.js

```diff
@@ -0,0 +1,50 @@
+import pg from "pg"
+
+const pool = new pg.Pool({
+    connectionString: "postgres://postgres:password@localhost:5432/pets-app"
+})
+
+class SurrenderApplication{
+  constructor({id = null, name, phoneNumber, phone_number, email, petName, petAge,    petType, petImage, vaccinationStatus = "No"}){
+    //surrender_application fields
+    this.id = id;
+    this.name = name;
+    this.phoneNumber = phone_number || phoneNumber;
+    this.email = email;
+    this.status = "Pending";
+    //This needs to be set at record insert time, after the adoptable_pet is created
+    this.adoptable_pet_id = null;
+    //adoptable_pets fields
+    this.petName = petName;
+    this.img_url = petImage;
+    this.age = petAge;
+    this.pet_type = petType === "fish"? petType: petType + "s";
+    this.vaccination_status = vaccinationStatus || vaccination_status;
+    this.adoption_story = "Not available";
+  }
+
+  async save(){
+    try {
+      //Get the pet_type ID
+      const query0 = "SELECT id FROM pet_types WHERE type = $1;"
+      let result = await pool.query(query0, [this.pet_type]);
+      let petTypeId = result.rows[0].id;
+      //Create the adoptable pet and get it's id.
+      const query1 = "INSERT INTO adoptable_pets (name, img_url, age, vaccination_status, adoption_story, pet_type_id) VALUES ($1, $2, $3, $4, $5, $6) RETURNING id;"
```

Comment:
This save method should only create the surrender application because that is the model for this file. The adoptable pets model should have it's own save method

Reply:
So, we are not allowed to access tableB from ModelA even though we must access tableB to save modelA?

Reply:
I aslo have to access tableC before I can access tableB. If I access thes tables via ModelB and ModelC, I've still accessed tableB and tableC via tableA.

Reply:
Just in an around the world fashion.

Reply:
Should I only access those Models via their respective Routers and
should I only access those Routers via the view that needs access to them?